org.alloytools.alloy.application/src/main/java/templates/TestTemplateFile.vm

```diff
@@ -533,7 +533,10 @@ class CentralController:
                 if possible_transitions:
                     chosen_transition = random.randint(0, len(possible_transitions) - 1)
                     possible_transitions[chosen_transition]()
-                    # if conc_state's active state changed (i.e. conc_state took a transition), prevent additional transitions (big step maximality)
+                    # prevent the current state from taking additional transitions if it is conc (big step maximality)
+                    if cur_state.is_conc:
+                        unused_conc_states.remove(cur_state)
+                    # if any other conc_state's active state changed (i.e. the conc_state took a transition), prevent additional transitions
```

Comment:
Let's say I have a conc state A and it has an active child OR state B. Let's say B takes a transition that does not have a goto clause. 
1. The `if cur_state.is_conc:` condition wouldn't be triggered because B is not a conc state
2. The `if conc_state.get_active_child_state() != conc_state_to_active_child[conc_state]:` wouldn't be triggered either because the active child state of A is still B?
So there is still a case where we would violate big set maximality?

Reply:
I don't think so. As far as I'm aware, there's no restriction on the number of transitions an OR state can take in a big step.

Reply:
I think we have different understandings of the concept
The original wording is "at most one transition per concurrent state can be taken in a big step"
If an OR state inside a conc state takes a transition, my understanding is that the conc state is also treated as taken a transition. But let me ask in slack which understanding is correct.

Reply:
Also looks like this counts the destination state of the transition? I am not sure if we need to count destination state as taken a transition.